components/Wizard.vue

```diff
@@ -0,0 +1,308 @@
+<script>
+/*
+Wizard accepts an array of steps (see props), and creates named slots for each step.
+It also contains slots for buttons:
+    next
+    cancel
+    finish
+
+Wizard will emit these events:
+    next({step})
+    cancel
+    finish
+*/
+
+export default {
+  props:      {
+    mode: {
+      type:    String,
+      default: 'create'
+    },
+
+    /*
+  steps need: {
+    name: String - this will be the slot name
+    label: String - this will appear in the top nav bar below circles
+    subText: String (optional) - If defined, appears in the Step {number}: banner. If blank, label is used
+    ready: Boolean - whether or not the step is completed/wizard is able to go to next step
+      if a step has ready=true, the wizard also allows navigation *back* to it
+  }
+  */
+    steps: {
+      type:    Array,
+      default: () => []
+    },
+
+    // Initial step to show when Wizard loads. This is overruled by the 'step' query param, if available
+    initStepIdx: {
+      type:    Number,
+      default: 0
+    },
+
+    // if true, allow users to navigate back to the first step of the Wizard
+    // if false, only way back to step 1 is to cancel and undo all configuration
+    editFirstStep: {
+      type:    Boolean,
+      default: false
+    }
+  },
+
+  asyncData(ctx) {
+    const { route:{ query } } = ctx;
+
+    return { queryStep: query.step };
+  },
+
+  data() {
+    return { activeStep: this.queryStep || this.steps[this.initStepIdx] };
+  },
+
+  computed: {
+    activeStepIndex() {
+      return this.steps.indexOf(this.activeStep);
+    },
+
+    canNext() {
+      return (this.activeStepIndex < this.steps.length - 1) && this.activeStep.ready;
+    },
+
+    readySteps() {
+      return this.steps.filter(step => step.ready);
+    }
+
+  },
+
+  watch: {
+    '$route.query'(neu = {}) {
+      this.goToStep(neu.step);
+    },
+  },
+
+  created() {
+    const { $route: { query = {} } } = this;
+
+    // if any step has active===true, load that step. Otherwise, load step defined in query param. Otherwise, load first step.
+    const stepNumber = this.activeStep ? this.steps.indexOf(this.activeStep) + 1 : query.step || 1;
+
+    const step = this.steps[stepNumber - 1] || this.steps[0];
+
+    if ( step ) {
+      this.goToStep(stepNumber);
+    }
+  },
+
+  methods: {
+    goToStep(number, fromNav) {
+      // if editFirstStep is false, do not allow returning to step 1 (restarting wizard) from top nav
+      if (!this.editFirstStep && (number < 1 || (number === 1 && fromNav))) {
+        return;
+      }
+
+      const {
+        steps,
+        $route: { query:{ step: queryStep } },
+      } = this;
+
+      const selected = steps[number - 1];
+
+      if ( !selected || !this.isAvailable(selected)) {
+        return;
+      }
+
+      if (queryStep !== number) {
+        this.$router.push({ ...this.$route, query: { ...this.$route.query, step: number } });
```

Comment:
Use `STEP` from `config/query-params` everywhere instead of literal string so usage is easier to find.

 [applyQuery](https://github.com/rancher/dashboard/blob/master/plugins/extend-router.js#L4) is a shorter equivalent of this, and supports having lack of a query param for the "default" value of step=1:

```javascript
this.$router.applyQuery({[STEP]: number}, {[STEP]: 1})
```